python/make_finemap_inputs.py

```diff
@@ -143,6 +146,20 @@ def read_sumstats(path,
     return sumstats
 
 
+def merge_regions(regions, max_region_width):
```

Comment:
@Fedja I got your point re: restricting region size; but I'm not sure whether the current implementation is a good approach... Since `regions` are sorted by genomic positions, the current implementation might mess up the following situation:

When `max_region_width` = 5 Mb, the current implementation would generate 1) 1-6 Mb and 2) 4-7Mb but it'd be better to have 1) 1-4 Mb and 2) 2-7 Mb given the last region has the smallest p-value.

```
chromosome	start	end	lead_p
1	1000000	4000000	5e-10
1	2000000	5000000	5e-20
1	3000000	6000000	5e-30
1	4000000	7000000	1e-100
```

In practice though, it might not matter much... Given we know there is much better implementaion (e.g. LD independent regions) anyway, this might be sufficient for now. What do you think?

Reply:
hmmm yea true that.  in practice it does not matter much. especially if setting the region large to begin with but maybe better approach actually could be to:
1) merge all regions as traditionally
2) if size exceeds threshold, then take the top SNP in the region and get max region size around that define that as region and remove from the big region
3) repeat 2 until the region 1 is split to regions smaller than max size

How do you think that might work?

and yea LD version is the way to go. PRobably could quickly implement LD based region definition using FinnGen LD API if just increasing the max region size that spits out.   Too late for r6 obvi but that could probably work quite nicely.... in practice none of this matters for 97% of cases as 3MB is large enough.

Reply:
Yeah, I think your proposed approach would work better! Okay, given we have r6 fine-mapping with this implementation, I accept this PR for now as a record. We can further develop on this in future releases.

Reply:
no that is actually worse in some situations as we can end up with narrow region for some signals in the ends of the regions.... aaargh

Reply:
ah I thought your step 2 could exceed the edges of the original merged region so that even a small region at the edge would have at least 3 Mb

Reply:
yea but it could be too short on the other side

Reply:
and we don't want to have overlapping regions finemapped